classes/class04/burgerAppHw/public/javascripts/main.js

```diff
@@ -0,0 +1,166 @@
+
+//GENERIC onError FUNCTION
+var onError = function(data, status) {
+	console.log("status", status);
+	console.log("error", data);
+};
+
+
+
+
+//HANDLING OUT OF STOCK
+var $form = $(".edit_form");
+var onSuccessOut = function(data, status) {
+	$currentForm = $("#"+data._id);
+	$currentForm.remove();
+	console.log(data);
+};
+
+$form.submit(function outOfStock(event){
+	event.preventDefault();
+	var postData = {id:$(this).attr("id")};
+	$.post("ingredients", postData)
+		.done(onSuccessOut)
+		.error(onError);
+});
+
+
+//RESOLVING AN ORDER
+var $removeOrdBut = $(".removeOrder");
+var onSuccessDelOrd = function(data, status){
+	$currentLI = $("#"+data._id);
+	$currentLI.remove();
+	console.log(data);
+}
+$removeOrdBut.click(function resolveOrder(event){
+	event.preventDefault();
+	var postData = {id: $(this).parent().attr("id")};
+	
+	$.post("delOrd", postData)
+		.done(onSuccessDelOrd)
+		.error(onError);
+});
+
+
+//EDIT NAME OF AN INGREDIENT
+var $editNameBut = $(".editName");
+var onSuccessName = function(data,status){
+	$currentForm = $("#"+data._id);
+	editedName = data.name;
+	var newName = $currentForm.find("span.nameSpan").html("Name: "+ editedName);
+
+}
+$editNameBut.click(function editName(event){
+	event.preventDefault();
+	var postData = {name: $(this).parent()
```

Comment:
This has some weird layout. Consider breaking up into multiple lines to make it display nicer.